contracts/bots/PartialLiquidationBot.sol

```diff
@@ -0,0 +1,280 @@
+/// @dev It is expected that this is set as a special permission bot in BotListV3 for all Credit Managers
+contract PartialLiquidationBot is IPartialLiquidationBot {
+    /// @notice Minimal health factor of account after liquidation
+    uint256 public constant THRESHOLD_HEALTH_FACTOR = 10200;
+
+    /// @notice The percentage of total debt that can be covered by partial liquidaitons,
+    ///         accounting for possible future interest
+    uint256 public constant DEBT_BUFFER = 10100;
+
+    /// @notice Performs a partial liquidation by swapping some CA collateral to underlying at a discount.
+    ///         Accepts the amount of underlying that the liquidator is willing to spend
+    /// @param creditManager Credit Manager where the account resides
+    /// @param creditAccount Credit Account to liquidate
+    /// @param assetOut Asset to receive for underlying
+    /// @param amountIn Amount of underlying to pay. Note that depending on the CA's balance in assetOut and
+    ///                 the maximal swappable amount of underlying, the actual spent amount may be less
+    /// @param repay Whether to repay debt or only perform a swap
+    /// @param priceUpdates Data for updatable price feeds, if any are required to compute account health
+    function partialLiquidateExactIn(
+        address creditManager,
+        address creditAccount,
+        address assetOut,
+        uint256 amountIn,
+        bool repay,
+        PriceUpdate[] memory priceUpdates
+    ) external returns (uint256, uint256) {
+        LiquidationParams memory params =
+            _prepareParams(creditManager, creditAccount, assetOut, amountIn, 0, repay, true);
+
+        /// Since we are computing debt and collateral before liquidation,
+        /// we need to update prices beforehand, if needed
+        _applyPriceUpdates(priceUpdates, params.priceOracle);
+
+        _retrieveAndCheckDebt(params);
+
+        _liquidate(params);
+
+        return (params.amountIn, params.amountOut);
+    }
+
+    /// @notice Performs a partial liquidation by swapping some CA collateral to underlying at a discount.
+    ///         Accepts the amount of assetOut that the liquidator wishes to receive
+    /// @param creditManager Credit Manager where the account resides
+    /// @param creditAccount Credit Account to liquidate
+    /// @param assetOut Asset to receive for underlying
+    /// @param amountOut Amount of assetOut to receive. Note that depending on the CA's balance in assetOut and
+    ///                 the maximal swappable amount of underlying, the actual received amount may be less
+    ///                 (with the spent amount adjusted accordingly)
+    /// @param repay Whether to repay debt or only perform a swap
+    /// @param priceUpdates Data for updatable price feeds, if any are required to compute account health
+    function partialLiquidateExactOut(
```

Comment:
Looks like we can combine `partialLiquidateExactIn` and `partialLiquidateExactOut` and only add a single flag `exactIn` since the code is way too similar to keep two functions

Reply:
Unfortunately, the function ends up having too many params and causes a "stack too deep" error

Reply:
One way to overcome it is to combine `amountIn` and `amountOut` args of `_prepareParams` to just `amount` (the other is always zero) and dispatch there based on `exactIn` flag:

```solidity
LiquidationParams memory params = 
    _prepareParams(creditManager, creditAccount, assetOut, amount, to, repay, exactIn);
```

Reply:
Tried this and a few other ways, still hitting the stack limit for me